util/url.go

```diff
@@ -0,0 +1,31 @@
+package util
+
+import (
+	"net/url"
+
+	"github.com/goccy/go-json"
+	"github.com/pkg/errors"
+)
+
+// URL is a wrapper struct of url.URL in order to unmarshal(marshal) URLs from(to) strings.
+type URL struct {
+	url.URL
+}
+
+func (u URL) MarshalJSON() ([]byte, error) {
```

Comment:
I believe you should be able to just call this Marshal and it will implement the json marshal interface

Reply:
looks like you had it right my bad http://choly.ca/post/go-json-marshalling/